code/game/objects/structures/window.dm

```diff
@@ -245,7 +250,7 @@
 			playsound(src, 'sound/items/Welder.ogg', 80, 1)
 			construction_state = 0
 			set_anchored(0)
-	else
+	else if (!istype(W, /obj/item/device/paint_sprayer))
```

Comment:
`else if (!istype(W, /obj/item/paint_sprayer))`
We don't have `/device` type now.